File: LeetCodeProblemsGo/LeetCode2942FindWordsContainingCharacter/LeetCode2942FindWordsContainingCharacter.go
package main

func findWordsContaining(words []string, x byte) []int {
	length := len(words)
	result := []int{}
	runeX := rune(x)
	for i := 0; i < length; i++ {
		for _, c := range words[i] {
			if c == runeX {
				result = append(result, i)
				break
			}
		}
	}
	return result
}

File: LeetCodeProblemsCMake/Problems/LeetCode2942FindWordsContainingCharacter.cpp
//
// Created by saeed on 5/24/25.
//

#include "LeetCode2942FindWordsContainingCharacter.h"
using namespace std;

vector<int> LeetCode2942FindWordsContainingCharacter::findWordsContaining(vector<string> &words, char x) {
    vector<int> result{};
    int length=static_cast<int>(words.size());
    for(int i=0;i<length;i++) {
        for (auto c:words[i]) {
            if (c == x) {
                result.push_back(i);
                break;
            }
        }
    }
    return result;
}
